airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/OracleSourceComprehensiveTest.java

```diff
@@ -0,0 +1,253 @@
+
+  private OracleContainer container;
+  private JsonNode config;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(OracleSourceComprehensiveTest.class);
+
+  @Override
+  protected Database setupDatabase() throws Exception {
+    container = new OracleContainer("epiclabs/docker-oracle-xe-11g");
+    container.start();
+
+    config = Jsons.jsonNode(ImmutableMap.builder()
+        .put("host", container.getHost())
+        .put("port", container.getFirstMappedPort())
+        .put("sid", container.getSid())
+        .put("username", container.getUsername())
+        .put("password", container.getPassword())
+        .build());
+
+    Database database = Databases.createOracleDatabase(config.get("username").asText(),
+        config.get("password").asText(),
+        String.format("jdbc:oracle:thin:@//%s:%s/%s",
+            config.get("host").asText(),
+            config.get("port").asText(),
+            config.get("sid").asText()));
+    LOGGER.warn("config: " + config);
+
+    database.query(ctx -> ctx.fetch("CREATE USER test IDENTIFIED BY test DEFAULT TABLESPACE USERS QUOTA UNLIMITED ON USERS"));
+
+    return database;
+  }
+
+  @Override
+  protected String getIdColumnName() {
+    return "ID";
+  }
+
+  @Override
+  protected String getTestColumnName() {
+    return "TEST_COLUMN";
+  }
+
+  @Override
+  protected String getNameSpace() {
+    return "TEST";
+  }
+
+  @Override
+  protected void tearDown(TestDestinationEnv testEnv) {
+    container.close();
+  }
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/source-oracle:dev";
+  }
+
+  @Override
+  protected JsonNode getConfig() {
+    return config;
+  }
+
+  @Override
+  protected void initTests() {
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("CHAR")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .fullSourceDataType("CHAR(3 CHAR)")
+            .addInsertValues("null", "'a'", "'ab'", "'abc'")
+            .addExpectedValues(null, "a  ", "ab ", "abc")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("VARCHAR2")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .fullSourceDataType("VARCHAR2(256)")
+            .addInsertValues("null", "'тест'", "'⚡ test ��'", "q'[{|}!\"#$%&'()*+,-./:;<=>?@[]^_`~]'")
+            .addExpectedValues(null, "тест", "⚡ test ��", "{|}!\"#$%&'()*+,-./:;<=>?@[]^_`~")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("VARCHAR2")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .fullSourceDataType("VARCHAR2(256)")
+            .addInsertValues("chr(33) || chr(34) || chr(35) || chr(36) || chr(37) || chr(38) || chr(39) || chr(40) || chr(41)")
+            .addExpectedValues("!\"#$%&'()")
+            .build());
+
+    addDataTypeTestData(
```

Comment:
NCHAR  type can be added to verification within max lengs and zero length value. The same for NVARCHAR2